library/src/main/java/com/novoda/downloadmanager/lib/BatchQuery.java

```diff
@@ -0,0 +1,223 @@
+                        + "ELSE 2 "
+                        + "END",
+                Downloads.Impl.STATUS_RUNNING,
+                Downloads.Impl.STATUS_PENDING,
+                Downloads.Impl.STATUS_PAUSED_BY_APP,
+                Downloads.Impl.STATUS_BATCH_FAILED,
+                Downloads.Impl.STATUS_SUCCESS
+        );
+
+        private Criteria.Builder criteriaIdBuilder;
+        private Criteria.Builder criteriaStatusBuilder;
+
+        public Builder() {
+            builder = new Criteria.Builder();
+        }
+
+        /**
+         * Filter by batch id
+         *
+         * @param id id of the batch
+         * @return {@link BatchQuery.Builder}
+         */
+        public Builder withId(long id) {
+            this.criteriaIdBuilder = new Criteria.Builder();
+            criteriaIdBuilder
+                    .withSelection(Downloads.Impl.Batches._ID, Criteria.Wildcard.EQUALS)
+                    .withArgument(String.valueOf(id));
+            return this;
+        }
+
+        /**
+         * Sorting in ascending order by sort column
+         *
+         * @param sortColumn sort column
+         * @return {@link BatchQuery.Builder}
+         */
+        public Builder withSortAscendingBy(String sortColumn) {
+            builder.sortBy(sortColumn).ascending();
+            return this;
+        }
+
+        /**
+         * Sorting in descending order by sort column
+         *
+         * @param sortColumn sort column
+         * @return {@link BatchQuery.Builder}
+         */
+        public Builder withSortDescendingBy(String sortColumn) {
+            builder.sortBy(sortColumn).descending();
+            return this;
+        }
+
+        /**
+         * Sorts batches according to the 'liveness' of the download, i.e. in the order:
+         * Downloading, queued, other, paused, failed, completed
+         *
+         * @return {@link BatchQuery.Builder}
+         */
+        public Builder withSortByLiveness() {
+            builder.sortBy(ORDER_BY_LIVENESS);
+            return this;
+        }
+
+        /**
+         * Filter by status
+         *
+         * @param statusFlags status flags that can be combined with "|"
+         *                    one of {@link DownloadManager#STATUS_PAUSED},
+         *                    {@link DownloadManager#STATUS_FAILED},
+         *                    {@link DownloadManager#STATUS_PENDING},
+         *                    {@link DownloadManager#STATUS_SUCCESSFUL},
+         *                    {@link DownloadManager#STATUS_RUNNING}
+         *                    <p/>
+         *                    e.g. withStatusFilter(DownloadManager.STATUS_FAILED | DownloadManager.STATUS_PENDING)
+         * @return {@link BatchQuery.Builder}
+         */
+        public Builder withStatusFilter(@Status int statusFlags) {
+            this.criteriaStatusBuilder = new Criteria.Builder()
+                    .or(buildCriteriaListFrom(statusFlags));
```

Comment:
Not completely sure about this though, related discussion [here](https://github.com/novoda/download-manager/pull/79#discussion_r34183529)